index.js

```diff
@@ -141,31 +153,27 @@ Transformer.prototype.compile = function (str, options) {
   return tr.normalizeFn(this._tr.compile(str, options));
 };
 Transformer.prototype.compileAsync = function (str, options, cb) {
-  if (!this.can('compileAsync')) {
-    if (this.can('compileFileAsync')) {
-      return Promise.reject(new Error('The Transform "' + this.name + '" does not support compiling plain strings')).nodeify(cb);
-    } else {
-      return Promise.reject(new Error('The Transform "' + this.name + '" does not support compilation')).nodeify(cb);
-    }
+  if (!this.can('compileAsync')) { // compileFile* || renderFile* || renderAsync || compile*Client*
```

Comment:
Should we also add a `.renderAsync` fallback for `.compileAsync`? Or perhaps after we add this one?